index.js

```diff
@@ -4,7 +4,18 @@ const Router = require('call').Router;
 const EventEmitter = require('events');
 const helpers = require('./helpers');
 const INTERPOLATION_REGEX = /\{([^}]*)\}/g;
-
+async function sendMessage(socket, message) {
+    if (socket.readyState === ws.OPEN) {
+        if (socket.auth) {
+            try {
+                await helpers.jwtXsrfCheck(socket.auth);

```

Comment:
Checking whole token is not optimal, please cache and check only the expiration property of the decoded token.